src/typechecker/FStar.TypeChecker.Rel.fs

```diff
@@ -934,6 +946,17 @@ let occurs_check (uk:ctx_uvar) t =
                         (Print.uvar_to_string uk.ctx_uvar_head)
                         (Print.term_to_string t)) in
     uvars, not occurs, msg
+let occurs_full (uk:ctx_uvar) t =
```

Comment:
This computes the occurs check while accounting for ctx_uvar_typ, i.e., it uses uvars_full, but this function is never called currently. I would like to experiment with what it takes to make this function the default occurs check